mlflow2seldon/api.py

```diff
@@ -0,0 +1,288 @@
+from __future__ import annotations
+import os
+import sys
+import re
+import signal
+import logging
+import tempfile
+import yaml
+import subprocess
+import asyncio
+from pathlib import Path
+from dataclasses import dataclass
+from typing import Any, Dict
+
+from yarl import URL
+from neuro_sdk import Factory, Client
+from mlflow.tracking.client import MlflowClient
+from mlflow.exceptions import MlflowException
+
+
+DELAY = 2
+CLUSTER_FROM_JOB_URL_MASK = "jobs.(.*).org.neu.ro"
+
+
+@dataclass
+class _DeployedModel:
+    image: str
+    model_name: str
+    model_storage_uri: URL
+    model_stage: str
+    model_version: str
+    source_run_id: str
+    deployment_namespace: str
+    need_redeploy: bool = False
+
+    def is_same_version(self, other: _DeployedModel) -> bool:
+        all_is_true = all(
+            (
+                self.name == other.name,
+                self.model_version == other.model_version,
+            )
+        )
+        return all_is_true
+
+    @property
+    def name(self) -> str:
+        return f"{self.model_name}-{self.model_stage}".lower()
+
+
+async def poll_mlflow(env: Dict):
+
+    # Settings of the source cluster, where MLflow is deployed:
+    mlflow_storage_root = env["M2S_MLFLOW_STORAGE_ROOT"]
+    mlflow_host = env["M2S_MLFLOW_HOST"]
+    default_image = env["M2S_SELDON_NEURO_DEF_IMAGE"]
+    deploy_image_tag = env["M2S_MLFLOW_DEPLOY_IMG_TAG"]
+    registry_secret_name = env["M2S_NEURO_REGISTRY_SECRET"]
```

Comment:
if we separate variables `M2S_MLFLOW_` and `M2S_SELDON_`, we'll need to specify which neuro cluster this is for